src/accelerate/checkpointing.py

```diff
@@ -241,6 +246,11 @@ def load_accelerator_state(
             sampler = dataloader.get_sampler()
             if isinstance(sampler, SeedableRandomSampler):
                 sampler = dataloader.set_sampler(torch.load(input_sampler_file))
+        if getattr(dataloader, "use_stateful_dataloader", False):
+            dataloader_state_dict_name = "dl_state_dict.bin" if i == 0 else f"dl_state_dict_{i}.bin"
```

Comment:
Probably there is no way for this to happen, but I just wanted to bring it up: Could there be a situation where this condition is entered from an older checkpoint, where the dataloader checkpoint does not exist yet, thus leading to  a crash?